addons/account/models/account_account.py

```diff
@@ -737,20 +737,21 @@ def _order_accounts_by_frequency_for_partner(self, company_id, partner_id, move_
         return self._get_most_frequent_accounts_for_partner(company_id, partner_id, move_type)
 
     @api.model
-    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
-        if not name and self._context.get('partner_id') and self._context.get('move_type'):
-            return self._order_accounts_by_frequency_for_partner(
-                            self.env.company.id, self._context.get('partner_id'), self._context.get('move_type'))
-        domain = domain or []
-        if name:
-            if operator in ('=', '!='):
-                name_domain = ['|', ('code', '=', name.split(' ')[0]), ('name', operator, name)]
-            else:
-                name_domain = ['|', ('code', '=like', name.split(' ')[0] + '%'), ('name', operator, name)]
-            if operator in expression.NEGATIVE_TERM_OPERATORS:
-                name_domain = ['&', '!'] + name_domain[1:]
-            domain = expression.AND([name_domain, domain])
-        return self._search(domain, limit=limit, order=order)
+    def _search_display_name(self, operator, value):
+        if not value and self._context.get('partner_id') and self._context.get('move_type'):
+            ids = self._order_accounts_by_frequency_for_partner(
+                self.env.company.id, self._context.get('partner_id'), self._context.get('move_type'))
+            return [('id', 'in', ids)]
```

Comment:
These lines seem to be for the public name_search instead? 

It is coming from https://github.com/odoo/odoo/pull/91938

Reply:
It was inside method `_name_search`.

Reply:
I wanted to leave as-is since it is based on the context. If it were only for the public web, it should have been written in `name_search` instead of `_name_search`. I want to minimize the changes where I am not sure.